### ISSUE TITLE
Support window.zoomLevel as workspace setting

### ISSUE THREAD

**Issue by mlibre** [Microsoft/vscode#38177]
- VSCode Version: 1.18.0
- OS Version: linux. debian 8. steam os

Steps to Reproduce:

1.open vscode

Reproduces without extensions: Yes.

no matter what  "window.zoomLevel": is.
vscode dose not read it. and i have to change it to different value so it will work.
it was fine before vscode 1.18.
![vscode](https://user-images.githubusercontent.com/8473036/32708872-f3d98d34-c841-11e7-92c1-1de49c082995.gif)

**Comment by bpasero** [comment]
@mlibre it looks like the settings file is dirty and not saved when you start, is that always the case?

**Comment by mlibre** [comment]
Yeh.  I forgot about that. but i just did it for gif. still not fixed.
setting file: [settings.txt](https://github.com/Microsoft/vscode/files/1469888/settings.txt)
new gif:
![vscode-2](https://user-images.githubusercontent.com/8473036/32768052-6258bd86-c92b-11e7-86af-8dd14f8dbe8e.gif)

**Comment by mlibre** [comment]
adding **"window.zoomLevel": 1** to "**user setting**" solved the problem.
so there is something wrong with reading configs from "**workspace setting**".

**Comment by bpasero** [comment]
I can reproduce when window.zoomLevel is a workspace setting. This might be hard to fix but I will investigate.

**Comment by bpasero** [comment]
Turns out we do not support this for quite a while. One of the main reasons is that we read the setting very early on the main side where we do not have a configuration service that has access to any workspace settings.

@sandy081 is there any way I can mark a setting as "global" only? This would actually be useful for a number of settings (like disabling auto update).

**Comment by mlibre** [comment]
@bpasero Thank you. This is a rly nice feature. I preserve all of my settings in **workspace** so i can easily move project + setting to somewhere else. and finally I have the same IDE **everywhere**. this feature increase **protability**.

**Comment by sandy081** [comment]
@bpasero 

@mlibre says that it is working in before 1.17 and I confirm it. Looks like a regression.

Need to investigate, what caused this.

**Comment by sandy081** [comment]
Yes, its a regression that caused due to the fine grained configuration event adoption. We always had a configuration change event listener that updates the window zoom level by comparing the current zoomLevel value webFrame zoomLevel value. This is called when the workbench is started in the renderer when the default configurations are registered. With new configuration change event, I adopted that listener to only listen to changes to `window.ZoomLevel` that caused this issue.

Now, I separated out listening and acting part and it is much cleaner now.
